go/client/cmd_simplefs_list.go

```diff
@@ -95,28 +198,58 @@ func (c *CmdSimpleFSList) Run() error {
 		if err != nil {
 			return err
 		}
+		gotList := false
 		for {
 			listResult, err := cli.SimpleFSReadList(ctx, opid)
+			// Eat the error here because it may just mean the results
+			// are complete. TODO: should KBFS return non-error here
+			// until the opid is closed?
 			if err != nil {
-				break
+				if gotList == true {
+					err = nil
+				}
+				return err
+			}
+			gotList = true
+			err = c.output(listResult)
+			if err != nil {
+				return err
 			}
-			c.output(listResult)
 		}
 	}
 	return err
 }
 
-func (c *CmdSimpleFSList) output(listResult keybase1.SimpleFSListResult) {
-
+func (c *CmdSimpleFSList) output(listResult keybase1.SimpleFSListResult) error {
 	ui := c.G().UI.GetTerminalUI()
 
-	for _, e := range listResult.Entries {
-		if e.DirentType == keybase1.DirentType_DIR {
-			ui.Printf("%s\t<%s>\t\t%s\n", keybase1.FormatTime(e.Time), keybase1.DirentTypeRevMap[e.DirentType], e.Name)
-		} else {
-			ui.Printf("%s\t%s\t%d\t%s\n", keybase1.FormatTime(e.Time), keybase1.DirentTypeRevMap[e.DirentType], e.Size, e.Name)
+	if c.winStyle {
+		for _, e := range listResult.Entries {
+			if e.DirentType == keybase1.DirentType_DIR {
+				ui.Printf("%s\t<%s>\t\t%s\n", keybase1.FormatTime(e.Time), keybase1.DirentTypeRevMap[e.DirentType], e.Name)
+			} else {
+				ui.Printf("%s\t%s\t%d\t%s\n", keybase1.FormatTime(e.Time), keybase1.DirentTypeRevMap[e.DirentType], e.Size, e.Name)
+			}
+		}
+	} else {
+		// capture the current terminal dimensions
+		terminalWidth, _, err := terminal.GetSize(int(os.Stdout.Fd()))
+		if err != nil {
+			return err
+		}
+
+		var outputBuffer bytes.Buffer
+
+		err = c.ls(&outputBuffer, listResult, terminalWidth)
+		if err != nil {
+			return err
+		}
+
+		if outputBuffer.String() != "" {
+			ui.Printf("%s", outputBuffer.String())
```

Comment:
I did check, it does not. I made this `ui.Printf("%s -- end", outputBuffer.String())` and got:
```
c:\work\src\github.com\keybase\client\go\keybase>keybase fs ls -l /keybase/private/zanderz
-rw-rw-r--      116460896 Mar 03 10:47 Keybase_1.0.18-20170302120457+22e9a36.386.exe
-rw-rw-r--        4097907 Feb 27 10:35 Peter Berkow - Practice 2 - Purple Hair.mp3
-rw-rw-r--            736 Feb 08 18:16 setup.bat
-rw-rw-r--            436 Feb 08 18:16 sshnotes.txt
drw-rw-r--            920 Mar 08 10:24 test
-rw-rw-r--             27 Feb 28 13:10 test4.txt
-rw-rw-r--            618 Feb 08 18:16 testkeys.txt
 -- end
c:\work\src\github.com\keybase\client\go\keybase>
```

so it really seems like Keybase itself adds a newline. But we could shave off the last newline if you want.